### PR TITLE
Real time Voice Chat Demo

### LINKED ISSUE #15: Feature Request: Add Real-time Voice Chat Demo
## Overview
It would be great to have a demo that enables natural voice conversations with omni-mini-2 using WebRTC technology. This would allow users to interact with the model through speech in real-time, both locally and in cloud environments.

## Existing Implementation Reference
I already have a working implementation for omni-mini-1 on Hugging Face Spaces that demonstrates this capability:

https://github.com/user-attachments/assets/040ec236-b336-4a14-880a-f9cde82e63e1

## Key Benefits
- Real-time voice interactions
- Works both locally and in cloud/remote environments
- Enhanced user experience through natural conversation

## Technical Resources
- Reference implementation for omni-mini-1: [Source Code](https://huggingface.co/spaces/freddyaboulton/omni-mini-webrtc/blob/main/app.py)
- WebRTC component documentation: [gradio-webrtc](https://github.com/freddyaboulton/gradio-webrtc)

## Potential Implementation
The port from omni-mini-1 to omni-mini-2 should be relatively straightforward given the existing implementation. This could be a valuable addition to enhance the project's interactive capabilities.

Would this be a feature the maintainers are interested in adding to the project?

### PR DESCRIPTION
**Pull request by freddyaboulton**
Closes #15 

Modifies the gradio demo to allow real time conversation both locally and on deployed cloud environments.

https://github.com/user-attachments/assets/440c385c-5da3-4f91-81f3-462e2fe3209a

Demo running on spaces: https://huggingface.co/spaces/freddyaboulton/mini-omni2-webrtc

I changed the UI slightly after this recording but you get the idea 😄

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -12,8 +12,9 @@ streamlit-webrtc
...
 onnxruntime==1.19.0
 # numpy==1.26.3
-gradio==4.42.0
 librosa==0.10.2.post1
 flask==3.0.3
 fire
-git+https://github.com/mini-omni/CLIP.git
\ No newline at end of file
+git+https://github.com/mini-omni/CLIP.git
+gradio_webrtc[vad]==0.0.11
```
Comment (freddyaboulton) [inline requirements.txt:19]: gradio is a dependency of gradio_webrtc. By default this will install gradio 5 which is secure and more performant.

**Comment by mini-omni** [comment]
Great, thanks for your great contribution, I will test it ASAP.

**Comment by mini-omni** [comment]
@freddyaboulton Hi, my testing works fine locally and runs well. It seems the Gradio version needs to be 5.0 or above. Also, could you please explain how to run this on a remote server? I'm having issues accessing the mic due to HTTPS problems. If I use share=True, it can generate an HTTPS link, but the speed is very slow.

**Comment by freddyaboulton** [comment]
Hi @mini-omni !

> Also, could you please explain how to run this on a remote server?

For best results, sign up for the [twilio api](https://www.twilio.com/docs/stun-turn). It is free and you won't be charged. Then create the `rtc_configuration` like [this](https://github.com/gpt-omni/mini-omni2/pull/27/files#diff-2cbef08d4331ba197c71a8c58fc5eb0ba564729e3897eb87d9e47a3c70a786c7R37) and pass it to the WebRTC component. Would not recommend using share links as the data would have to go through an intermediate server.

**Pull request merged**